### PR TITLE
Add general node pools page

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/giantswarm/issues/5465 https://github.com/giantswarm/giantswarm/issues/7075

This PR adds a page at https://docs.giantswarm.io/basics/nodepools/ that explains all the basics about Node Pools.

### CODE REVIEW

**Comment by marians** [comment]
FYI: This branch has a link to a gsctl reference page that is currently still missing. It's coming with a different PR.

**Comment by marians** [comment]
@giantswarm/team-firecracker Thanks in advance for your reviews!

**Comment by marians** [comment]
@J-K-C If you would also focus on language here, that would help. Thanks!

File: src/content/basics/nodepools/index.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+Node pools can be created when creating a cluster
+
+- via the Giant Swarm web interface ("happa")
+- via the CLI command [`gsctl create cluster`](/reference/gsctl/create-cluster/)
+
+or any time after the cluster has been created
+
+- via the Giant Swarm web interface
+- via the CLI command [`gsctl add nodepool`](/reference/gsctl/add-nodepool/)
```
Comment (xh3b4sd) [inline src/content/basics/nodepools/index.md]: It is `create`, not `add`. Also, the link does not work, at least in the PR.

File: src/content/basics/nodepools/index.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+These tools also support modification of node pools and their deletion.
+
+Once a node pool has been created, as soon as the workers are available, they will
+join the cluster and appear in your `kubectl get nodes` listing. You can identify the
+nodes' node pool using the `giantswarm.io/machine-deployment` label.
+
+```nohighlight
+kubectl get nodes \
+  -o=jsonpath='{range .items[*]}{.metadata.labels.giantswarm\.io/machine-deployment}{"\t"}{.metadata.name}{"\n"}{end}' | sort
```
Comment (xh3b4sd) [inline src/content/basics/nodepools/index.md]: Example output would be really useful here.

File: src/content/basics/nodepools/index.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+```
+
+Some details of a node pool can be modified after creation:
+
+- The node pool name
+- The scaling range (min(max)
```
Comment (xh3b4sd) [inline src/content/basics/nodepools/index.md]: The braces are weird. Not sure if this is some well established format.

File: src/content/basics/nodepools/index.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+## Limitations
+
+- A node pool can have a maximum of 99 worker nodes.
+
+- At times, the EC2 instance type required by a node pool can be unavailable in certain availability zones. This can result in node pools providing less than the desired number of nodes. The more availability zones a node pool spans, the less likely this problem is to occur.
+
+- By default, clusters can have up to 5 node pools. This is limited by the AWS service quota named "IPv4 CIDR blocks per VPC" in the [VPC section](https://docs.aws.amazon.com/vpc/latest/userguide/amazon-vpc-limits.html). As the AWS account owner, you can request an increase of this limit via the AWS console.
+
+- Node pools can span a maximum of 4 different availability zones. This limit affects both the number of availability zones a single node pool can cover, as well as the number of different availability zones all node pools of a cluster can cover.
```
Comment (xh3b4sd) [inline src/content/basics/nodepools/index.md]: We need to make it clearer that AZs selected once cannot be undone. You have 4 AZ slots and once filled they are permanent for a cluster.

File: src/content/basics/nodepools/index.md (hunk 3 above)
Comment (marians) [reply src/content/basics/nodepools/index.md]: No, that's a typo. Will change to `(min/max)`.

File: src/content/basics/nodepools/index.md (hunk 4 above)
Comment (marians) [reply src/content/basics/nodepools/index.md]: Very good catch! Updated.

**Comment by marians** [comment]
PTAL @xh3b4sd - Made all the changes based on your feedback

File: src/content/basics/nodepools/index.md (hunk 2 above)
Comment (marians) [reply src/content/basics/nodepools/index.md]: Handled in https://github.com/giantswarm/docs/pull/363

**Pull request merged**